tests/webpub_manifest_parser/opds2/test_production_feeds.py

```diff
@@ -0,0 +1,98 @@
+import logging
+from unittest import TestCase, skip
+
+import requests
+from parameterized import parameterized
+
+from webpub_manifest_parser.core import ManifestParserResult
+from webpub_manifest_parser.opds2 import OPDS2FeedParserFactory
+
+
+class OPDS2FeedParserIntegrationTest(TestCase):
+    @parameterized.expand(
+        [
+            (
+                "Feedbooks",
+                "https://catalog.feedbooks.com/catalog/index.json",
+            ),
+            (
+                "fulcrum",
+                "https://www.fulcrum.org/api/opds/umpebc_oa",
+            ),
+        ]
+    )
+    def test_production_feed(self, _, url, encoding="utf-8"):
```

Comment:
The PR description mentions:
> OPDS 2.x: Feedbooks, Fulcrum, ProQuest (skipped by default - requires a running SOCKS proxy since the ProQuest feed is accessible only by whitelisted IP addresses).

The wording is ambiguous there. Feedbooks and Fulcrum are not skipped by default, only ProQuest, so it might be helpful to reword this to make it more clear.

Reply:
I updated the description.

Reply:
Regarding the skipping the test - this is what I'm not sure about. I would like to have these tests running all the time since they allow to understand if the parser is able to parse correct feeds. But at the same time those tests can give us false negatives: they can fail because of errors in the feeds or network issues.

Reply:
I skipped them.

Reply:
> Regarding the skipping the test - this is what I'm not sure about. I would like to have these tests running all the time since they allow to understand if the parser is able to parse correct feeds. But at the same time those tests can give us false negatives: they can fail because of errors in the feeds or network issues.

Exactly. I would say that those feeds represent "real" feeds, but not necessarily "correct" feeds. Additionally, what constitutes a "correct" feed may change over time as the spec changes.

My personal opinion is that these are separate problems:
- We can validate correct parsing of "correct" feeds with unit tests and a suite of correct feeds.
- We can assure that we are able to process the applicable "real" feeds with integration tests.

For us, applicable feeds would be the feeds that we use, but another user of this package might be using different feeds. As you've noted, another issue for the integration tests is the need for credentials. That might argue for situating these tests in an environment where we have these credentials.

Reply:
I tend to agree @tdilauro. I like the idea of storing a "correct" version of these feeds as a test fixture and running a test against it. 

Then as future work we could add a separate integration test action that runs integration tests against interesting feeds, so we can see if something changes in the integration that causes these tests to start failing. This could be either done here in github or in our hosting environment.

Reply:
@jonathangreen, @tdilauro, is that right that I can just grab a piece of Feedbooks and Fulcrum feeds and save those snippets in this repo? Won't it be a violation of any copyrights?

Reply:
I suspect that the feeds themselves are not copyrightable; though, that might not be true for the parts of them that involve some intellectual (non-factual) contribution (e.g., `.publications[].metadata.description`).

For what I called "correct" feeds above, I think there are a couple of ways forward:
- I don't think we need to use the exact content of specific feeds to create fixtures. We could, for example, change links.
- We could also seek permission to use them for this purpose.
- We could create our own or acquire sample feeds.

Dealing with validation of feeds in which we have specific production interest could be handled separately in integration tests. The integration would need to publish the feeds, so this would not be an issue.

Reply:
There is already a [snippet](https://github.com/ThePalaceProject/webpub-manifest-parser/blob/feature/odl2/tests/files/odl/feed.json) in the codebase. I took it from the ODL spec and it's quite illustrative.

Reply:
Yeah, that seems fine. We could update that or add more later, if needed.

Reply:
@jonathangreen, @tdilauro can we merge this PR?

Reply:
Looks good to me as well.